src/lib/server/orderTab.ts

```diff
@@ -93,46 +97,94 @@ export async function concludeOrderTab({ slug }: { slug: string }) {
 	await collections.orderTabs.deleteOne({ slug });
 }
 
-export async function checkoutOrderTab({ slug, user }: { slug: string; user: UserIdentifier }) {
+export async function checkoutOrderTab({
+	slug,
+	user,
+	splitMode,
+	itemQuantities
+}: {
+	slug: string;
+	user: UserIdentifier;
+	splitMode?: 'shares' | 'items';
+	itemQuantities?: Map<string, number>;
+}) {
 	const orderTab = await getOrCreateOrderTab({ slug });
-	const cartItems = orderTab.items.map((line) => ({
-		_id: line._id.toString(),
-		productId: line.productId,
-		quantity: line.quantity,
-		internalNote: line.internalNote,
-		chosenVariations: line.chosenVariations
-	}));
+
+	// Build cart items
+	const cartItems = itemQuantities
+		? Array.from(itemQuantities.entries()).map(([itemId, qty]) => {
+				const tabItem = orderTab.items.find((i) => i._id.toString() === itemId);
+				if (!tabItem) {
+					const availableItems = orderTab.items.map((i) => i._id.toString());
+					console.error('Item not found in orderTab:', {
+						requestedItemId: itemId,
+						availableItems
+					});
+					throw error(400, 'Item not found in order tab');
+				}
+				return {
+					_id: itemId,
+					productId: tabItem.productId,
+					quantity: qty,
+					internalNote: tabItem.internalNote,
+					chosenVariations: tabItem.chosenVariations
+				};
+		  })
+		: orderTab.items.map((line) => ({
+				_id: line._id.toString(),
+				productId: line.productId,
+				quantity: line.quantity,
+				internalNote: line.internalNote,
+				chosenVariations: line.chosenVariations
+		  }));
+
+	// Determine final splitMode
+	const finalSplitMode = itemQuantities ? 'items' : splitMode;
```

Comment:
It's very counter-intuitive that `splitMode` will be ignored if `itemQuantities` is specified.

Can you improve the function typing to make this obvious?